app/src/main/java/com/team7/cmput301/android/theirisproject/activity/PatientListActivity.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Team 7, CMPUT301, University of Alberta - All Rights Reserved. You may use, distribute, or modify this code under terms and conditions of the Code of Students Behavior at University of Alberta
+ *
+ *
+ */
+
+package com.team7.cmput301.android.theirisproject.activity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.DialogFragment;
+import android.view.View;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import com.team7.cmput301.android.theirisproject.PatientListAdapter;
+import com.team7.cmput301.android.theirisproject.R;
+import com.team7.cmput301.android.theirisproject.controller.PatientListController;
+import com.team7.cmput301.android.theirisproject.model.Patient;
+import com.team7.cmput301.android.theirisproject.task.Callback;
+
+import java.util.List;
+
+/**
+ * PatientListActivity is the screen which shows a Care Provider all their registered Patients.
+ * It lets them add a new Patient by clicking the add button, which uses AddPatientFragment to help
+ * add the Patient by their email
+ *
+ * @see AddPatientFragment
+ * @author Jmmxp
+ */
+public class PatientListActivity extends IrisActivity<List<Patient>> implements AddPatientFragment.AddPatientDialogListener {
+
+    private PatientListController controller;
+    private ListView patientsView;
+    private PatientListAdapter adapter;
+
+    @Override
+    protected PatientListController createController(Intent intent) {
+        return new PatientListController(intent);
+    }
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_patient_list);
+
+        controller = createController(getIntent());
+        patientsView = findViewById(R.id.patient_item_list);
+
+        // set click listener to AddProblemFloatingButton
+        findViewById(R.id.patient_list_add).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                DialogFragment addPatientDialog = new AddPatientFragment();
+                addPatientDialog.show(getSupportFragmentManager(), AddPatientFragment.class.getSimpleName());
```

Comment:
I'm assuming the when a dialog is shown, the current activity is paused. If so, when we exit dialog you are making an async call to the database but also onStart() will do the same. Thus there is a race condition which I think is why you are not getting the updated list of patients

Reply:
This might not be the problem since I'm re-calling "render" again when the Patient is finished being added in "onFinishAddPatient", I believe the problem and the similar problem in your ProblemListActivity is because we need to be calling adapter.notifyDatasetChanged(), but I will add this as an issue so we can have the CareProvider functionality in master to add tests and such first